GPU/GLES/GLES_GPU.cpp

```diff
@@ -1915,15 +1928,18 @@ void GLES_GPU::Execute_Generic(u32 op, u32 diff) {
 }
 
 void GLES_GPU::FastLoadBoneMatrix(u32 target) {
+	const int num = gstate.boneMatrixNumber & 0x7F;
+	const int mtxNum = num / 12;
+	uint32_t uniformsToDirty = DIRTY_BONEMATRIX0 << mtxNum;
+	if ((num - 12 * mtxNum) != 0) {
+		uniformsToDirty |= DIRTY_BONEMATRIX0 << ((mtxNum + 1) & 7);
+	}
+
 	if (!g_Config.bSoftwareSkinning || (gstate.vertType & GE_VTYPE_MORPHCOUNT_MASK) != 0) {
 		Flush();
-		const int num = gstate.boneMatrixNumber & 0x7F;
-		int mtxNum = num / 12;
-		uint32_t uniformsToDirty = DIRTY_BONEMATRIX0 << mtxNum;
-		if ((num - 12 * mtxNum) != 0) {
-			uniformsToDirty |= DIRTY_BONEMATRIX0 << ((mtxNum + 1) & 7);
-		}
 		shaderManager_->DirtyUniform(uniformsToDirty);
+	} else {
+		gstate_c.deferredVertTypeDirty |= uniformsToDirty;
```

Comment:
No, we exactly cannot do that.

Example:

1. There is no morph.
2. The game updates a bone matrix.  Soft skinning does its magic.
3. Morph is turned on.  It was off before.
4. Draw happens.  Reuses previously set bone matrix.
5. Draw must have CORRECT bone matrix.

We have to compute the dirty flags always.  The only case we can avoid them is when we are sure that morph will **never** be used without first updating the bone matrices it references or turning off skinning.

-[Unknown]